frontend/recepies.generated.online/src/components/postcard-stuff/Paypal.vue

```diff
@@ -2,85 +2,78 @@
-                                value: this.amount
+    export default {
+        props: ['recipeID', 'sendTo', 'amount'],
+        methods: {
+            initPayPalButton() {
+                let _this = this
+                paypal.Buttons({
+                    style: {
+                        shape: 'rect',
+                        color: 'black',
+                        layout: 'vertical',
+                        label: 'paypal',
+                        tagline: false
+                    },
+                    createOrder: (data, actions) => {
+                        this.$analytics.logEvent("paypal_created_order");
+                        return actions.order.create({
+                            //intent: 'CAPTURE', // DEV ONLY
+                            "application_context": {
+                                shipping_preference: "SET_PROVIDED_ADDRESS",
                             },
-                            shipping: {
-                                name: {
-                                    full_name: this.sendTo.name
+                            purchase_units: [{
+                                description: this.recipeID + " an " + this.sendTo.name +
+                                    " von " + this.sendTo.absender,
+                                amount: {
+                                    currency_code: "EUR",
+                                    value: this.amount
                                 },
-                                address: {
-                                    address_line_1: this.sendTo.street,
-                                    postal_code: this.sendTo.plz,
-                                    admin_area_2: this.sendTo.city,
-                                    country_code: this.sendTo.country
+                                shipping: {
+                                    name: {
+                                        full_name: this.sendTo.name
+                                    },
+                                    address: {
+                                        address_line_1: this.sendTo.street,
+                                        postal_code: this.sendTo.plz,
+                                        admin_area_2: this.sendTo.city,
+                                        country_code: this.sendTo.country
+                                    }
                                 }
-                            }
-                        }],
-                    });
-                },
+                            }],
+                        });
+                    },
+                    onApprove: function (data, actions) {
+                        _this.$analytics.logEvent("paypal_order_worked");
+                        return actions.order.capture().then(function (details) {
+
+                            let db = firebase.firestore();
+                            db.collection("orders").doc().set({
+                               "sendTo": this.sendTo,
```

Comment:
i think you cannot reference this here, you need to change it to _this
```suggestion
                               "sendTo": _this.sendTo,
```